### PR TITLE
feat(lang): add zig support

### PR DESCRIPTION
**Pull request by Thoxy67**
Hello there! 

This extra allows Zig to work seamlessly within LazyVim. 
To further improve its functionality, we can consider adding debugging. 
However, I'm currently unfamiliar with how to integrate it with Zig. 

On the bright side, both LSP and Neotest are working like a charm

### CODE REVIEW

File: lua/lazyvim/plugins/extras/lang/zig.lua [hunk 1]
```diff
@@ -0,0 +1,68 @@
+return {
+  {
+    "NTBBloodbath/zig-tools.nvim",
+    ft = "zig",
+    config = function()
```
Comment (dpetka2001) [inline lua/lazyvim/plugins/extras/lang/zig.lua:5]: I suggest you use `opts = {}` or `config = true`. This way the user can pass his own changes to the `opts`, without any need to fiddle with the `config` function. Also, by using `config` like this, you're overriding the default `config` function and since you don't pass any `opts` to the new `config` function, the user would have to pass all the settings in the `setup` table of the `config` function in his own configuration.

Comment (Thoxy67) [reply lua/lazyvim/plugins/extras/lang/zig.lua]: I have made the change for opts [49803c17accb133ac29d9357f1bf46512b3b65c2](https://github.com/LazyVim/LazyVim/pull/1750/commits/49803c17accb133ac29d9357f1bf46512b3b65c2)

File: lua/lazyvim/plugins/extras/lang/zig.lua [hunk 2]
```diff
@@ -2,8 +2,8 @@ return {
   {
     "NTBBloodbath/zig-tools.nvim",
     ft = "zig",
-    config = function()
-      require("zig-tools").setup()
+    opts = function(_, opts)
+      require("zig-tools").setup(opts)
```
Comment (dpetka2001) [inline lua/lazyvim/plugins/extras/lang/zig.lua:6]: Why are you trying to `require` the module? That's already what `lazy.nvim` does. A simple `opts = {}` should do it I believe (since setting `opts` implies the call to `Plugin.config` which calls the setup function of the module). In my opinion, this is redundant.

Comment (dpetka2001) [reply lua/lazyvim/plugins/extras/lang/zig.lua]: No need to worry. I wasn't criticizing you or anything. Just pointed out how it could be simplified, since you weren't customizing the `config` function yourself and the plugin doesn't need some specific need of setup, like for example `nvim-treesitter` which needs to be setup by doing `require'nvim-treesitter.configs'.setup` and `lazy.nvim` wouldn't be able to call the correct module on its own based on the heuristics it uses to identify the correct `main` module. Keep up the good work. I'm just a newbie like you. `lazy.nvim`'s docs are a treasure providing great knowledge and I myself have to revisit them from time to time to clarify things, so I suggest you give them a good read to understand better how things work.

File: lua/lazyvim/plugins/extras/lang/zig.lua [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    "NTBBloodbath/zig-tools.nvim",
+    ft = "zig",
+    opts = {},
+    dependencies = {
+      {
+        "akinsho/toggleterm.nvim",
+      },
+      {
+        "nvim-lua/plenary.nvim",
+        module_pattern = "plenary.*",
```
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/zig.lua:12]: `module_pattern` is not a valid property of a spec

File: lua/lazyvim/plugins/extras/lang/zig.lua [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+  {
+    "NTBBloodbath/zig-tools.nvim",
+    ft = "zig",
+    opts = {},
+    dependencies = {
+      {
+        "akinsho/toggleterm.nvim",
+      },
+      {
+        "nvim-lua/plenary.nvim",
```
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/zig.lua:11]: plenary is already part of LAzyVim, no need to specify it here

File: lua/lazyvim/plugins/extras/lang/zig.lua [hunk 5]
```diff
@@ -0,0 +1,66 @@
+return {
+  {
+    "NTBBloodbath/zig-tools.nvim",
+    ft = "zig",
+    opts = {},
+    dependencies = {
+      {
+        "akinsho/toggleterm.nvim",
```
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/zig.lua:8]: I'm not a fan of pulling in toggleterm just for zig. It also is not setup, so will cause issues

File: lua/lazyvim/plugins/extras/lang/zig.lua [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      },
+      {
+        "nvim-lua/plenary.nvim",
+        module_pattern = "plenary.*",
+      },
+    },
+  },
+  {
+    "ziglang/zig.vim",
```
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/zig.lua:17]: why is this needed if we have treesitter?

File: lua/lazyvim/plugins/extras/lang/zig.lua [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    end,
+  },
+  {
+    "neovim/nvim-lspconfig",
+    opts = {
+      servers = {
+        zls = {},
+      },
+    },
+    setup = {
```
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/zig.lua:44]: this should be under `lspconfig.opts`, not root of the spec

File: lua/lazyvim/plugins/extras/lang/zig.lua [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+      end,
+    },
+  },
+  {
+    "nvim-neotest/neotest",
+    optional = true,
+    dependencies = {
+      "lawrence-laz/neotest-zig",
+      "nvim-lua/plenary.nvim",
+      "nvim-treesitter/nvim-treesitter",
```
Comment (folke) [inline lua/lazyvim/plugins/extras/lang/zig.lua:58]: these deps are not needed

File: lua/lazyvim/plugins/extras/lang/zig.lua (hunk 5 above)
Comment (Thoxy67) [reply lua/lazyvim/plugins/extras/lang/zig.lua:8]: If i remove toggleterm zig-tools give me this error every time (no config needed to prevent this error) :

```
   Error  17:50:10 notify.error lazy.nvim Failed to run `config` for zig-tools.nvim

...e/LazyVim/lazy/zig-tools.nvim/lua/zig-tools/commands.lua:14: module 'toggleterm.terminal' not found:
	no field package.preload['toggleterm.terminal']
cache_loader: module toggleterm.terminal not found
cache_loader_lib: module toggleterm.terminal not found
	no file './toggleterm/terminal.lua'
	no file '/usr/share/luajit-2.1/toggleterm/terminal.lua'
	no file '/usr/local/share/lua/5.1/toggleterm/terminal.lua'
	no file '/usr/local/share/lua/5.1/toggleterm/terminal/init.lua'
	no file '/usr/share/lua/5.1/toggleterm/terminal.lua'
	no file '/usr/share/lua/5.1/toggleterm/terminal/init.lua'
	no file './toggleterm/terminal.so'
	no file '/usr/local/lib/lua/5.1/toggleterm/terminal.so'
	no file '/usr/lib/lua/5.1/toggleterm/terminal.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'
	no file './toggleterm.so'
	no file '/usr/local/lib/lua/5.1/toggleterm.so'
	no file '/usr/lib/lua/5.1/toggleterm.so'
	no file '/usr/local/lib/lua/5.1/loadall.so'

# stacktrace:
  - /zig-tools.nvim/lua/zig-tools/commands.lua:14
  - /zig-tools.nvim/lua/zig-tools/autocmds.lua:8
  - /zig-tools.nvim/lua/zig-tools.lua:9
  - /usr/share/nvim/runtime/filetype.lua:31
  - /usr/share/nvim/runtime/filetype.lua:30
  - vim/_editor.lua:0
  - /telescope.nvim/lua/telescope/actions/set.lua:200 _in_ **run_replace_or_original**
  - /telescope.nvim/lua/telescope/actions/mt.lua:65 _in_ **run_replace_or_original**
  - /telescope.nvim/lua/telescope/actions/mt.lua:65 _in_ **run_replace_or_original**
  - /telescope.nvim/lua/telescope/actions/mt.lua:65 _in_ **key_func**
  - /telescope.nvim/lua/telescope/mappings.lua:257

```

Comment (Conarius) [reply lua/lazyvim/plugins/extras/lang/zig.lua:8]: From another similar pr, zig-tools.vim could also be used as it's also by the authors of zig themselves and doesn't have toggleterm as a hard dependency. But, as you may guess, it's written in Vimscript, not in Lua

**Comment by folke** [comment]
like I said, toggleterm is a no go
